pkg/migration/controllers/migration.go

```diff
@@ -1320,6 +1319,24 @@ func (m *MigrationController) checkAndUpdateDefaultSA(
 			destSA.ImagePullSecrets = append(destSA.ImagePullSecrets, s)
 		}
 	}
+
+	for _, s := range sourceSA.Secrets {
+		if !m.secretToBeMigrated(s.Name) {
```

Comment:
Is this required as the secrets with the prefixes in AutoCreatedPrefixes are anyway not collected in resourcecollector?

Reply:
Secrets with AutoCreatedPrefixes are not collected or migrated. But if service has an associated secret that isn't auto created, it will be migrated and we should migrate it's reference in SA too.